lib/cyclonedx/report_ruby_gems_cyclonedx.rb

```diff
@@ -3,5 +3,20 @@ class ReportRubyGems < Base
     def initialize(scan_report)
       super(scan_report)
     end
+
+    def package_url(dependency)
+      "pkg:#{dependency[:type]}/#{dependency[:name]}#{version_string(dependency, true)}"
+    end
+
+    # Return version string to be used in purl
+    def version_string(dependency, is_purl_version = false)
+      prefix = is_purl_version ? "@" : ""
```

Comment:
Minor suggestion.  A guard class up top is preferred to the conditional if else block / flows:

```
return "" unless dependency[:dependency_file] == 'Gemfile.lock'
prefix = is_purl_version ? "@" : ""
"#{prefix}#{dependency[:version]}"
```

Reply:
Nice, will make the change 👍